app/forms/user_phone_form.rb

```diff
@@ -62,8 +61,11 @@ def otp_delivery_preference_changed?
     otp_delivery_preference != phone_configuration&.delivery_preference
   end
 
-  def update_otp_delivery_preference_for_user
-    user_attributes = { otp_delivery_preference: otp_delivery_preference }
+  def update_user
+    user_attributes = { remember_device_revoked_at: Time.zone.now }
+    if otp_delivery_preference.present? && otp_delivery_preference_changed?
```

Comment:
But see #2594

Reply:
Ah, fun. I'll let Moncef merge that and then I'll come back around and clean this one up.